File: app/src/main/java/com/example/gymlog/data/plan/PlanCycle.java
package com.example.gymlog.data.plan;

import java.util.List;

public class PlanCycle {
    private long id; // Унікальний ідентифікатор програми
    private String name; // Назва програми
    private String description; //опис
    private List<GymDay> gymDays; // Список тренувальних днів

    // Конструктори
    public PlanCycle() {}

    public PlanCycle(long id, String name, String description, List<GymDay> gymDays) {
        this.id = id;
        this.name = name;
        this.gymDays = gymDays;
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public String getDescription() {
        return description;
    }
}

File: app/src/main/java/com/example/gymlog/database/room/PlanCycle.kt
package com.example.gymlog.database.room

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey


@Entity(tableName = "PlanCycles")
data class PlanCycle(
    @PrimaryKey(autoGenerate = true) val id: Long? = null,
    val name: String,                       // NOT NULL
    val description: String?,
    val creation_date: String?,             // nullable
    val position: Int?,
    @ColumnInfo(defaultValue = "0")         // <‑‑ Room тепер очікує DEFAULT 0
    val is_active: Int? = 0
)
